spec/system/nested_imageable_spec.rb

```diff
@@ -0,0 +1,264 @@
+require "rails_helper"
+
+describe "Nested imageable" do
+  factories = [
+    :budget,
+    :budget_investment,
+    :future_poll_question_option,
+    :proposal
+  ]
+
+  let(:factory) { factories.sample }
+  let!(:imageable) { create(factory) }
+  let!(:user) { create(:user, :level_two) }
+  let(:path) do
+    if factory == :budget
+      new_admin_budgets_wizard_budget_path
+    elsif factory == :budget_investment
+      [
+        new_budget_investment_path(budget_id: imageable.budget_id),
+        new_management_budget_investment_path(budget_id: imageable.budget_id)
+      ].sample
+    elsif factory == :future_poll_question_option
+      new_admin_option_image_path(option_id: imageable.id)
+    elsif factory == :proposal
+      [new_proposal_path, edit_proposal_path(imageable)].sample
+    end
+  end
+  let(:submit_button_text) do
+    if factory == :budget
+      "Continue to groups"
+    elsif factory == :budget_investment
+      "Create Investment"
+    elsif factory == :future_poll_question_option
+      "Save image"
+    elsif factory == :proposal
+      if edit_path?(path)
+        "Save changes"
+      else
+        "Create proposal"
+      end
+    end
+  end
+  let(:notice_text) do
+    if factory == :budget
+      "New participatory budget created successfully!"
+    elsif factory == :budget_investment
+      "Budget Investment created successfully."
+    elsif factory == :future_poll_question_option
+      "Image uploaded successfully"
+    elsif factory == :proposal
+      if edit_path?(path)
+        "Proposal updated successfully"
+      else
+        "Proposal created successfully"
+      end
+    end
+  end
+
+  context "New and edit path" do
+    before do
+      create(:administrator, user: user) if admin_section?(path) || management_section?(path)
+      imageable.update!(author: user) if edit_path?(path)
+      do_login_for(user, management: management_section?(path))
```

Comment:
Maybe we don't need to pass the `path` parameter to these methods, since it's the result of a `let` block :thinking:.

```suggestion
      create(:administrator, user: user) if admin_section? || management_section?
      imageable.update!(author: user) if edit_path?
      do_login_for(user, management: management_section?)
```